cell_classification/segmentation_data_prep_test.py

```diff
@@ -33,9 +49,165 @@ def test_get_image():
         assert not np.array_equal(CD8_img, CD4_img)
 
 
+def prepare_test_data_folders(
+    num_folders, temp_dir, selected_markers, random=False, scale=1.0
+):
+    data_folders = []
+    for i in range(num_folders):
+        folder = os.path.join(temp_dir, "fov_1" + str(i))
+        os.mkdir(folder)
+        data_folders.append(folder)
+        for marker in selected_markers:
+            if random:
+                img = np.random.rand(256, 256) * scale
+            else:
+                img = np.ones([256, 256])
+            imwrite(
+                os.path.join(temp_dir, "fov_1" + str(i), marker + ".tiff"),
+                img,
+            )
+    return data_folders
+
+
+def test_calculate_normalization_matrix():
+
+    # instantiate data_prep, conversion_matrix and markers
+    data_prep = prep_object()
+    conversion_matrix = prepare_conversion_matrix()
+    selected_markers = list(conversion_matrix.columns)
+    selected_markers.remove("cluster_labels")
+
+    # check normalization_dict for different stochastic images
+    for scale in [0.5, 9.132]:
+
+        # create temporary folders with data and do tests
+        with tempfile.TemporaryDirectory() as temp_dir:
+            data_folders = prepare_test_data_folders(
+                5, temp_dir, selected_markers, random=True, scale=scale
+            )
+            norm_dict = data_prep.calculate_normalization_matrix(
+                data_folders=data_folders,
+                normalization_dict_path=os.path.join(temp_dir, "norm_dict_test.json"),
+                normalization_quantile=0.99,
+                selected_markers=selected_markers,
+            )
+
+            # check if the normalization_dict has the correct values for stochastic images
```

Comment:
Because the normalization value for each marker here is the same, this test would pass if the function had been written to take whatever normalization quantile was found for the first marker, and apply it to all of the other markers. 

We should check that when different markers have different normalization values, the function accurately returns them

Reply:
This one still needs to be addressed